pkg/daemon/daemon.go

```diff
@@ -426,3 +425,70 @@ func (d *daemon) DeletePeriodicUpdate() {
 
 	glog.Info("DeletePeriodicUpdate(): finished")
 }
+
+//  InitPool check the guids that are already allocated by the running pods
+func (d *daemon) InitPool() error {
+	glog.Info("InitPool():")
```

Comment:
I believe the correct approach is to let GUID pool track allocated GUIDs by first initializing it with a list of allocated GUIDs then adding/removing GUIDs from the pool as needed.

in that way, there is no need for `guidPodNetworkMap` in the Daemon.
the only issue i see with this approach is with `AllocateGUID()`.

@moshe010 @Mmduh-483 In which flow we risk allocating the same GUID for two pods or two networks on the same pod as depicted in line #472?

Reply:
so user can provide a guid in the pod spec. (like it done with mac address)
In that case we just want to allocate the requested guid if he do not provided we generate the guid.

Reply:
@moshe010 @adrianchiris , In case the guid was allocated then the periodic add failed to do `smPlugin.AddGuidsToPkey` then we need to release the guid again.
The map guidPodNetworkMap was solution for that case, when allocated guid and for some reason the network fail e.g. (adding annotation, setting pkey, ...etc.) then keep it allocated for that guid then check it again in the next periodic update.

Reply:
ok keep it in this PR and we see if can remove it later

Reply:
There is a need to keep for each pod circling in the periodic job its state to reduce the number of calls to UFM.
another way of solving that (not related to this commit) is to keep a context for each pod, updating its processing state in the pipeline and skipping steps that where already done.

Reply:
in any case its not related to this commit, as for now we seem to need to have `guidPodNetworkMap`